### DISCUSSION TITLE
Dreame Z10 Pro sensor drifting

### DISCUSSION THREAD

**Discussion by JanKrAppel** [Hypfer/Valetudo#1884] [Q&A / Support]
I'm having a persistent issue with my Z10 Pro. After something like two or three months, the map rotates to the right a little more each time I let the vacuum clean. It appears that for some reason, the sensors drift just a little bit.

The issue can be "fixed" by resetting the map and doing a new mapping pass, but that messes up the associated segment IDs, so that I have to reassign them in Home Assistant. But that's a nuisance I don't want to go through every few months.

I know that this is most probably not a Valetudo issue, but does someone have similar symptoms or knows of a way to recalibrate or fix the sensors? I saw a similar symptom described in #1857 but the "fix" there seems to be a map reset as well.

I don't think the issue is caused by the base station rotating, because it looks like the rooms that I vacuum often drift more than the rooms I don't let the robot visit that often.

**Comment by sky321** [comment]
This is a know problem and I have the same on L10 too.

**Reply by Zealot2000** [reply]
I had a similar problem on my L10 Pro until I affixed the robot's base to the wall using double-sided sticky tape. This, and also never picking up the robot from anywhere except its base (preventing the need for repositioning) has reduced the map rotation/distortion considerably. There's still some bleeding of rooms into adjacent ones, but only a few pixels.
Also, make sure you're on the latest firmware (check https://robotinfo.dev/detail_dreame.vacuum.p2028_0.html)

**Reply by JanKrAppel** [reply]
Hm, that's disappointing, it severely limits how useful this robot is for home automation.

I'm running the latest firmware (1156, according to [dustbuilder](https://builder.dontvacuum.me/_dreame_p2028.html)). I fixed the base down to the floor with double sticky tape, maybe that helps a little bit.

**Comment by jmilljr24** [comment]
I've had the same problem with my z10 pro as well. The other suggestions are good. I also found that cleaning the sensors frequently and also running a full clean of the house (not just selected segments) on a regular basis has almost eliminated the problem for me. I can't remember the last time I had to reset my map.

**Comment by ChrM3r** [comment]
Hey,

I had the same issue and solved it somehow with that solution: https://www.reddit.com/r/Dreame_Tech/s/XSc1nxFjwa 
Also have a look at all the answers under that post

Short form: Write two little batch scripts to backup and restore the map data, extend the valetudo postboot scripts to setup a cron job on every boot (as the cron jobs are reseted when the robot restarts) that runs the restore script e.g. every week, month what ever.

With that hack I get a "fresh" map every sunday, and I "prevent" the map from getting rotated/shifted.
